components/gitpod-db/src/typeorm/migration/1630335270192-UpdateDevxMailAccordingToChangelog.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import {MigrationInterface, QueryRunner} from "typeorm";
+import { columnExists } from "./helper/helper";
+
+export class UpdateDevxMailAccordingToChangelog1630335270192 implements MigrationInterface {
+
+    public async up(queryRunner: QueryRunner): Promise<any> {
+        if (await columnExists(queryRunner, "d_b_user", "additionalData")) {
+            await queryRunner.query("update d_b_user set additionalData = json_set(additionalData, '$.emailNotificationSettings.allowsDevXMail', true) where json_extract(additionalData, '$.emailNotificationSettings.allowsChangelogMail') = true OR json_extract(additionalData, '$.emailNotificationSettings.allowsChangelogMail') = 'true' AND json_extract(additionalData, '$.emailNotificationSettings.allowsDevXMail') IS NULL")
```

Comment:
I believe `AND` takes precedence over `OR` [[1](https://dev.mysql.com/doc/refman/8.0/en/operator-precedence.html)], which wrongly leads to something like: 
```
WHERE 
  json_extract(additionalData, '$.emailNotificationSettings.allowsChangelogMail') = true 
  OR (
     json_extract(additionalData, '$.emailNotificationSettings.allowsChangelogMail') = 'true' 
     AND json_extract(additionalData, '$.emailNotificationSettings.allowsDevXMail') IS NULL
   )
```
Instead of:
```
WHERE 
  (
     json_extract(additionalData, '$.emailNotificationSettings.allowsChangelogMail') = true 
     OR json_extract(additionalData, '$.emailNotificationSettings.allowsChangelogMail') = 'true'
   )
   AND json_extract(additionalData, '$.emailNotificationSettings.allowsDevXMail') IS NULL
```

Reply:
Thanks, I only pushed this draft so that I could have a preview env to connect to the DB 🤣 I forgot people were already reviewing this PR.